src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -407,8 +415,16 @@ MetadataCreateCommands(void)
 	/* construct the foreign key constraints after all tables are created */
 	foreach_ptr(cacheEntry, propagatedTableList)
```

Comment:
There seems to be a lot of for loops in this method iterating the same list. The for loops are simple though (just an observation)

Reply:
It's mainly because of ordering (first do table creation tables for all tables, then do partitioning hierarchies for all tables, then do foreign keys for all tables), hence they should not be merged.